deploy/local_single_user_deploy.sh

```diff
@@ -237,123 +240,229 @@ stop_process() {
-    install
-    launch_process ${VISOR_BIN} "--config ${CONF_DIR}/visor_conf.json" "visor.log"
+start() {
+    echo -e "${INFO}Preparing to start ChronoLog...${NC}"
+    check_work_dir
+    mkdir -p "${MONITOR_DIR}"
+    mkdir -p "${OUTPUT_DIR}"
+    check_installation
+    generate_config_files ${NUM_KEEPERS} ${CONF_FILE} ${CONF_DIR} ${OUTPUT_DIR} ${NUM_GRAPHERS} ${MONITOR_DIR}
+
+    echo -e "${INFO}Starting ChronoLog...${NC}"
+    start_process ${VISOR_BIN} "--config ${CONF_DIR}/visor_conf.json" "visor.launch.log"
     sleep 2
     num_graphers=${NUM_GRAPHERS}
     for (( i=1; i<=num_graphers; i++ ))
     do
-        launch_process ${GRAPHER_BIN} "--config ${CONF_DIR}/grapher_conf_$i.json" "grapher_$i.log"
+        start_process ${GRAPHER_BIN} "--config ${CONF_DIR}/grapher_conf_$i.json" "grapher_$i.launch.log"
     done
     sleep 2
     num_keepers=${NUM_KEEPERS}
     for (( i=1; i<=num_keepers; i++ ))
     do
-        launch_process ${KEEPER_BIN} "--config ${CONF_DIR}/keeper_conf_$i.json" "keeper_$i.log"
+        start_process ${KEEPER_BIN} "--config ${CONF_DIR}/keeper_conf_$i.json" "keeper_$i.launch.log"
     done
-    sleep 2
-
-    echo -e "${DEBUG}Deployment done${NC}"
+    echo -e "${INFO}ChronoLog Started.${NC}"
 }
 
-kill() {
-    echo -e "${INFO}Killing ...${NC}"
-    kill_process ${KEEPER_BIN}
-    kill_process ${GRAPHER_BIN}
-    kill_process ${VISOR_BIN}
-    echo -e "${DEBUG}Kill done${NC}"
-}
-
-reset() {
-    echo -e "${INFO}Resetting...${NC}"
-    kill;
-    echo -e "${DEBUG}Delete all generated files${NC}"
-
-    # Remove all config files
-    rm -f ${CONF_DIR}/grapher_conf*.json
-    rm -f ${CONF_DIR}/keeper_conf*.json
-    rm -f ${CONF_DIR}/visor_conf.json
-
-    # Remove all monitor files
-    rm -f ${MONITOR_DIR}/*.log
-
-    # Remove all output files
-    rm -f ${OUTPUT_DIR}/*
-    echo -e "${DEBUG}Reset done${NC}"
-}
 
 stop() {
-    echo -e "${INFO}Stopping ...${NC}"
+    echo -e "${INFO}Stopping ChronoLog...${NC}"
+    check_work_dir
     declare -a processes=("${KEEPER_BIN}" "${GRAPHER_BIN}" "${VISOR_BIN}")
+
+    local start_time=$(date +%s)
+    local timeout=300  # 5 minutes in seconds
+
     for process in "${processes[@]}"; do
         stop_process "${process}"
+
+        # Wait for the process to stop with a timeout
         while pgrep -f "${process}" >/dev/null; do
             echo -e "${DEBUG}Waiting for ${process} to stop...${NC}"
             sleep 10
+
+            # Check if timeout is reached
+            local current_time=$(date +%s)
+            if (( current_time - start_time >= timeout )); then
+                echo -e "${ERROR}Stopping process timed out. Killing all processes.${NC}"
+                kill
+                return
+            fi
         done
         echo -e "${DEBUG}${process} stop done${NC}"
     done
-    echo -e "${DEBUG}All processes stopped${NC}"
+
+    echo -e "${INFO}ChronoLog stopped.${NC}"
+}
+
+
+clean() {
+    echo -e "${INFO}Cleaning ChronoLog...${NC}"
```

Comment:
should we force kill all processes just in case users did not stop before clean?

Reply:
No, I don't think we should kill any running processes as part of -clean. We should check if any processes are still running, report any if found and exit. 
if the user wants to stop/kill processes he should call -stop option explicitly

Reply:
agreed